Dockerfile

```
@@ -0,0 +1,6 @@
+FROM python:3.10-slim
+WORKDIR /code
+COPY requirements.txt .
+RUN pip3 install -r /code/requirements.txt --no-cache-dir
+COPY . /code
```

```suggestion
COPY . .
RUN pip3 install -r requirements.txt --no-cache-dir
```
После этого уже не нужно будет делать второй копи, так как скопируется всё в папку код. И строкой ниже нет смысла указыать абсолютный путь, так как команда воркдир создала папку код и перешла в неё

---

@AntonZelinsky 
в этом случае каждое минимальное изменение в коде влечет за собой постоянное выполнение pip install -r requirements, что заметно увеличивает время создания образа.

в моем случае pip install будет вызываться только если добавилась какая-то новая зависимость. Если поменялся только код без изменения зависимостей, то выполнятся только последние инструкции Dockerfile'а.

---

Т.к докер образ стромтся с помощью наложения слоёв с данными друг на друга, то у Жени сделано правильно. Объясню почему:
Файл requirements меняется гораздо реже, нежели основной код. Поэтому пересобирать слой с зависимостями так же часто, как и основные сорцы - бессмысленно и увеличивает время билда для образа

---

Ого, я не знал про это. 
То есть, если скопировали все файлы и хотя бы один файл проекта поменялся, то будут стягиваться все зависимости? А если скопировать сначала файл реквайрменты и вызвать установку из него зависимостей, то если файл реквайрмента не менялся, то ничего устанавливаться не будет?

Я всё правильно понял? Добавьте, пожалуйста, комментарии в этом файле, чтобы все тоже узнали про это.

---

Да, правильно понял. Могу даже скинуть про это видос обучающий)

---

@agamova добавь пожалуйста описание, почему так разделено копирование